scripts/ci/check-for-version-info-cells.py

```diff
@@ -0,0 +1,46 @@
+# This code is a Qiskit project.
+#
+# (C) Copyright IBM 2025.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Check all notebooks have exactly one `version-info` cell.
+"""
+
+from pathlib import Path
+import nbformat
+import sys
+
+guides_path = Path("./docs/guides")
+for notebook in guides_path.glob("*.ipynb"):
+    nb = nbformat.read(notebook, as_version=4)
+    num_version_info_cells = len(
+        list(
+            filter(
+                lambda cell: "version-info" in cell.metadata.get("tags", []),
+                nb.cells,
+            )
+        )
```

Comment:
Nit/FYI a list comprehension is more idiomatic in Python because it's seen as more readable

```suggestion
        [cell for cell in nb.cells if "version-info" in cell.metadata.get("tags", [])]
```